Backpack/Button/Classes/BPKButton.m

```diff
@@ -252,27 +322,101 @@ - (void)setColors {
     } else {
         [self setDisabledStyle];
     }
+
+    [self setNeedsDisplay];
 }
 
-- (void)setFont {
+- (void)updateFont {
     switch (self.size) {
         case BPKButtonSizeDefault: {
-            [self.titleLabel setFont:BPKFont.textBaseEmphasized];
+            if (!self.iconOnly) {
+                self.titleLabel.font = BPKFont.textSmEmphasized;
+            } else {
+                // NOTE: `intrinsicContentSize` takes the font into account
+                // even if there is not text hence we set it to 0
+                self.titleLabel.font = [UIFont systemFontOfSize:0];
+            }
             break;
         }
         case BPKButtonSizeLarge: {
-            [self.titleLabel setFont:BPKFont.textLgEmphasized];
+            if (!self.iconOnly) {
+                self.titleLabel.font = BPKFont.textLgEmphasized;
+            } else {
+                // NOTE: `intrinsicContentSize` takes the font into account
+                // even if there is not text hence we set it to 0
+                self.titleLabel.font = [UIFont systemFontOfSize:0];
+            }
             break;
         }
         default: {
             NSAssert(NO, @"Invalid size %d", (int)self.size);
             break;
         }
     }
+
+    [self setNeedsDisplay];
+}
+
+- (void)updateContentColor {
+    [self setTitleColor:self.currentContentColor forState:UIControlStateNormal];
+    self.imageView.tintColor = self.currentContentColor;
+    UIColor *highlightedContentColor;
+
+    switch (self.style) {
+        // Explicit fall-through
+        case BPKButtonStylePrimary:
+        case BPKButtonStyleFeatured:
+            highlightedContentColor = [self class].highlightedWhite;
+            break;
+        case BPKButtonStyleSecondary:
+            highlightedContentColor = [self class].highlightedBlue;
+            break;
+        case BPKButtonStyleDestructive:
+            highlightedContentColor = [self class].highlightedRed;
+            break;
```

Comment:
The pressed state of the Link buttons is a little odd as the icons are not greyed on press. Should there be a case in this switch block for Link style buttons?